tinygrad/renderer/wgsl.py

```diff
@@ -85,7 +88,8 @@ def render_kernel(self, function_name:str, kernel:list[str], bufs:list[tuple[str
     bind_it = iter(range(len(bufs)))
     external_local_bufs = [line.lstrip() for line in kernel if "var<workgroup>" in line]
     kernel[:] = [line for line in kernel if "var<workgroup>" not in line]
-    prg = "enable f16;\nfn nan() -> f32 { let bits = 0xffffffffu; return bitcast<f32>(bits); }\n"
+    prg = "enable f16;\n" if self.supports_f16 else ""
```

Comment:
Instead, why don't you check if f16 is in any uops? I think this is what we do for OpenCL.